core/providers/gemini.go

```diff
@@ -0,0 +1,1218 @@
+		return nil, err
+	}
+
+	providerName := provider.GetProviderKey()
+
+	// Validate input
+	if err := input.Validate(); err != nil {
+		return nil, newBifrostOperationError("invalid transcription input: "+err.Error(), err, providerName)
+	}
+
+	// Check file size limit (Gemini has a 20MB limit for inline data)
+	const maxFileSize = 20 * 1024 * 1024 // 20MB
+	if len(input.File) > maxFileSize {
+		return nil, newBifrostOperationError("audio file too large for inline transcription", fmt.Errorf("file size %d bytes exceeds 20MB limit", len(input.File)), providerName)
+	}
+
+	if input.Prompt == nil {
+		input.Prompt = StrPtr("Generate a transcript of the speech.")
+	}
+
+	// Prepare request body using shared function
+	requestBody := prepareGeminiGenerationRequest(input, params, nil)
+
+	// Use common request function
+	bifrostResponse, geminiResponse, bifrostErr := provider.completeRequest(ctx, model, key, requestBody, ":generateContent", params)
+	if bifrostErr != nil {
+		return nil, bifrostErr
+	}
+
+	// Extract text from Gemini response
+	var transcriptText string
+	if len(geminiResponse.Candidates) > 0 && geminiResponse.Candidates[0].Content != nil {
+		for _, p := range geminiResponse.Candidates[0].Content.Parts {
+			if p.Text != "" {
+				transcriptText += p.Text
+			}
+		}
+	}
+
+	// If no transcript text was extracted, return an error
+	if transcriptText == "" {
+		return nil, newBifrostOperationError("failed to extract transcript from Gemini response", fmt.Errorf("no transcript text found"), providerName)
+	}
+
+	// Extract usage metadata using shared function
+	inputTokens, outputTokens, totalTokens := extractGeminiUsageMetadata(geminiResponse)
+
+	// Update the response with transcription-specific data
+	bifrostResponse.Object = "audio.transcription"
+	bifrostResponse.Transcribe = &schemas.BifrostTranscribe{
+		Text: transcriptText,
+		Usage: &schemas.TranscriptionUsage{
+			Type:         "tokens",
+			InputTokens:  &inputTokens,
+			OutputTokens: &outputTokens,
+			TotalTokens:  &totalTokens,
+		},
+		BifrostTranscribeNonStreamResponse: &schemas.BifrostTranscribeNonStreamResponse{
+			Task:     StrPtr("transcribe"),
+			Language: input.Language,
+		},
+	}
+
+	if params != nil {
+		bifrostResponse.ExtraFields.Params = *params
+	}
+
+	return bifrostResponse, nil
+}
+
+func (provider *GeminiProvider) TranscriptionStream(ctx context.Context, postHookRunner schemas.PostHookRunner, model string, key schemas.Key, input *schemas.TranscriptionInput, params *schemas.ModelParameters) (chan *schemas.BifrostStream, *schemas.BifrostError) {
+	// Check if transcription stream is allowed for this provider
+	if err := checkOperationAllowed(schemas.Gemini, provider.customProviderConfig, schemas.OperationTranscriptionStream); err != nil {
+		return nil, err
+	}
+
+	providerName := provider.GetProviderKey()
+
+	// Validate input
+	if err := input.Validate(); err != nil {
```

Comment:
no need to validate input, is remove all validation code. All input validation is already done in bifrost.go